### ISSUE TITLE
schema contains type already defined in strawberry shake

### ISSUE THREAD

**Issue by DacoTaco** [ChilliCream/hotchocolate#5361]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

im currently using strawberry shake to generate a http client to communicate with a graphQL api.
in the schema a type 'Uri' is defined with the following : 
``` 
type Uri
{ 
	path: String!
}
```

however, compiling my project with this in my schema.graphql i get ```CSC : error SS0006: The name `Uri` was already registered by another type.```. I assume it is confusing it with System.Uri?

i have no control over the api's schema so im stuck with this and fixing it on my client's side.

### Steps to reproduce

1. 

### Product

Strawberry Shake

### Version

12.12.1

**Comment by michaelstaib** [comment]
No, the URI type is a standard scalar and I guess the collision is rather with the GraphQL scalar type `URI`.
The `SS0006` error code comes from the GraphQL analyzer. This would be a bug ... will have a look at this.

**Comment by DacoTaco** [comment]
eh? URI is a standard graphql scalar? i thought it was URL that was a standard scalar?

**Comment by DacoTaco** [comment]
could the stale tag be removed? or has this bug been fixed?

**Comment by michaelstaib** [comment]
Its not a standard type .... meant builtin.

**Comment by michaelstaib** [comment]
Can you provide a minimal repro?

**Comment by DacoTaco** [comment]
i will see if i can make a minimal example tomorrow, but in the meantime the problem was found in my ticketswap poller : https://github.com/DacoTaco/TicketSwapPoller/tree/main/TicketSwapPoller

**Comment by DacoTaco** [comment]
[TicketSwapPoller_bare_minimum.zip](https://github.com/ChilliCream/graphql-platform/files/10678352/TicketSwapPoller_bare_minimum.zip)

i have stripped the little code that was in there and have left everything about besides the basic schema + URI & something so it would generate the client.

in the real program a query uses the URI scalar

**Comment by DacoTaco** [comment]
@glen-84 : does the repo still compile?

**Comment by glen-84** [comment]
> @glen-84 : does the repo still compile?

No.

**Comment by DacoTaco** [comment]
> > @glen-84 : does the repo still compile?
> 
> No.

i just did a fresh git clone of the repo i linked above, and it build just fine. the bare minimum indeed doesn't build, but it also shows the error of the URI. 
i can't remember what i did for the bare minimum, but i suggest using the repo if you really want something buildable. to cause the issue you can find the Uri in schema.graphql and uncomment it.

**Comment by glen-84** [comment]
I've updated the `Steps to reproduce` section in the initial comment.

**Comment by michael-b-tt0** [comment]
Is there an elegant way around this issue without having to manually rename all the relevant entries in schema.graphql?
